### PR TITLE
feat(matrix): add more matrix ops

### PR DESCRIPTION
**Pull request by Marcos-cat**

### CODE REVIEW

File: lib.ua [hunk 1]
```diff
@@ -18,6 +18,16 @@ Median ← ÷2/+⊏⊟⊃⌊⌈÷2-1⧻.⊏⍏.
 
 # -- Matrices/Vectors --
 
+# Determinate
```
Comment (Omnikar) [inline lib.ua]:
```suggestion
# Determinant
```

File: lib.ua [hunk 2]
```diff
@@ -18,6 +18,16 @@ Median ← ÷2/+⊏⊟⊃⌊⌈÷2-1⧻.⊏⍏.
...
+MDet ← |1 ⍣(
+  °(¤¤)
+| /+ ⍜▽¯⋯2 ×≡(MDet⍉⊏) ⊙¤ ⍜⬚0≡°⊚¬ °⊏⍉:°⊂
+)
+# Matrix Cofactors
+MCofactors ← ≡(≡(MDet ↘1↻) ⊙¤°⊏ ⍉↘1↻)⊙¤°⊏
```
Comment (Omnikar) [inline lib.ua]: `⇡⧻⟜¤` might be more clear than `⊙¤°⊏`

File: lib.ua [hunk 3]
```diff
@@ -18,6 +18,16 @@ Median ← ÷2/+⊏⊟⊃⌊⌈÷2-1⧻.⊏⍏.
 
 # -- Matrices/Vectors --
 
+# Determinate
+MDet ← |1 ⍣(
+  °(¤¤)
+| /+ ⍜▽¯⋯2 ×≡(MDet⍉⊏) ⊙¤ ⍜⬚0≡°⊚¬ °⊏⍉:°⊂
```
Comment (Omnikar) [inline lib.ua]:
```suggestion
MDet ← |1 ⍣°(¤¤)(
  /+ ⍜▽¯⋯2 ×≡(MDet⍉⊏) ⊙¤ ⍜⬚0≡°⊚¬ °⊏⍉:°⊂
```

**Comment by Omnikar** [comment]
~~`MInv` is currently computing `MDet` twice, in the function and in `MCofactors`; which isn't exactly favorable. I suppose you could `memo` the determinant function, but I don't think that's the best solution since you're going to end up using a bunch of memory. So do you think you could refactor so that you don't end up computing the same determinant twice?~~

**Comment by Omnikar** [comment]
Oh hang on a sec, the two uses of `MDet` aren't computing the determinants of the same matrix are they… I may be stupid

**Comment by Omnikar** [comment]
I don't think the inversion is working correctly?
```uiua
[1_2
 3_4]
Mmp ⟜MInv
## ╭─       
## ╷ ¯5 ¯8  
##   ¯3 ¯5  
##         ╯
```
That doesn't exactly look like the identity matrix.

**Comment by Marcos-cat** [comment]
I will check that out, i could have sworn it was working before

**Comment by Marcos-cat** [comment]
I somehow messed up Cofactors so it doesn't work on even sized Matrices my bad, ill make a fix

File: lib.ua [hunk 4]
```diff
@@ -18,6 +18,14 @@ Median ← ÷2/+⊏⊟⊃⌊⌈÷2-1⧻.⊏⍏.
 
 # -- Matrices/Vectors --
 
+# Matrix Determinant
+MDet ← |1 ⍣°(¤¤)(/+×≡(MDet⍉⊏)⊙¤⍜⬚0≡°⊚¬°⊏⍉:⍜▽¯⋯2°⊂)
+Sq ←~ ⍜△[.√⊢]
+# Matrix Cofactors
+MCof ← ⍥¯⊞≠.◿2°⊏Sq≡(MDet Sq≡⊡⊙¤⊚¬⊞↥°⊟)⊙¤⬚0∵°⊚⊚⊸=.
```
Comment (Omnikar) [inline lib.ua]:
```suggestion
MSquarify ↚ ⍜△[.√⊢]
# Matrix Cofactors
MCof ← ⍥¯⊞≠.◿2°⊏MSquarify≡(MDet MSquarify≡⊡⊙¤⊚¬⊞↥°⊟)⊙¤⬚0∵°⊚⊚⊸=.
```
```suggestion
Sq ←~ ⍜△[.√⊢]
# Matrix Cofactors
MCof ← ⍥¯⊞≠.◿2°⊏Sq≡(MDet Sq≡⊡⊙¤⊚¬⊞↥°⊟)⊙¤⬚0∵°⊚⊚⊸=.
```
```suggestion
Sq ←~ ⍜△[.√⊢]
# Matrix Cofactors
MCof ← ⍥¯⊞≠.◿2°⊏Sq≡(MDet Sq≡⊡⊙¤⊚¬⊞↥°⊟)⊙¤⬚0∵°⊚⊚⊸=.
```

Comment (Omnikar) [reply lib.ua]: Not sure why this is displaying as three suggestions

**Comment by Omnikar** [comment]
Thanks

**Pull request merged**